src/pybind/mgr/dashboard/frontend/.eslintrc.json

```diff
@@ -0,0 +1,96 @@
+{
+  "root": true,
+  "ignorePatterns": [
+    "projects/**/*"
+  ],
+  "overrides": [
+    {
+      "files": [
+        "*.ts"
+      ],
+      "parserOptions": {
+        "project": [
+          "tsconfig.json",
+          "e2e/tsconfig.json"
+        ],
+        "createDefaultProgram": true
+      },
+      "extends": [
+        "plugin:@angular-eslint/ng-cli-compat",
+        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
+        "plugin:@angular-eslint/template/process-inline-templates"
+      ],
+      "rules": {
+        "quotes": [2, "single", {"avoidEscape": true}],
+        "@angular-eslint/component-selector": [
+          "error",
+          {
+            "type": "element",
+            "prefix": "cd",
+            "style": "kebab-case"
+          }
+        ],
+        "@angular-eslint/directive-selector": [
+          "error",
+          {
+            "type": "attribute",
+            "prefix": "cd",
+            "style": "camelCase"
+          }
+        ],
+        "@angular-eslint/no-attribute-decorator": "error",
+        "@angular-eslint/no-forward-ref": "error",
+        "@typescript-eslint/consistent-type-definitions": "error",
+        "@typescript-eslint/dot-notation": "off",
+        "@typescript-eslint/naming-convention": [
+          "warn",
+          {
+            "selector": "variable",
+            "format": ["camelCase"]
+          }
+        ],
+        "@typescript-eslint/explicit-member-accessibility": [
+          "off",
+          {
+            "accessibility": "explicit"
+          }
+        ],
+        // TODO: Change to error - From L58 to L67
+        "@typescript-eslint/no-empty-function": "error",
+        "@typescript-eslint/member-ordering": "warn", 
+        "no-underscore-dangle": "warn",
+        "@typescript-eslint/ban-types": "warn",
+        "prefer-arrow/prefer-arrow-functions": "warn",
+        "@typescript-eslint/consistent-type-assertions": "warn",
+        "@typescript-eslint/no-unused-expressions": "warn",
+        "max-len": "warn",
+        "@typescript-eslint/prefer-for-of": "warn",
+        "prefer-const": "warn",
```

Comment:
@epuertat @nizamial09 As these `eslint` rules involved a lot of changes, I thought to take it in a separate PR once this is migration is done so that we can focus on not breaking anything when doing them. Hence I have added a TODO comment and made the rules to warn us

Reply:
Make sense!